coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -303,21 +317,13 @@ static Schema joinSchemas(@Nonnull Schema leftSchema, @Nonnull Schema rightSchem
     Preconditions.checkNotNull(leftSchema);
     Preconditions.checkNotNull(rightSchema);
 
-    List<Schema.Field> combinedSchemaFields = new ArrayList<>();
-
-    for (Schema.Field field : leftSchema.getFields()) {
-      combinedSchemaFields
-          .add(new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultValue(), field.order()));
-    }
-
-    for (Schema.Field field : rightSchema.getFields()) {
-      combinedSchemaFields
-          .add(new Schema.Field(field.name(), field.schema(), field.doc(), field.defaultValue(), field.order()));
-    }
+    List<Schema.Field> combinedSchemaFields = new ArrayList<>(cloneFieldList(leftSchema.getFields()));
```

Comment:
Why create a new ArrayList?

Reply:
oh yeah thanks for catching - was using IDE's auto-fixing feature which added additional wrapper